wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/server/HttpBitbucketServerApiClient.java

```diff
@@ -227,7 +227,8 @@ public BitbucketPersonalAccessToken createPersonalAccessTokens(
               .PUT(
                   HttpRequest.BodyPublishers.ofString(
                       OM.writeValueAsString(
-                          new BitbucketPersonalAccessToken(tokenName, permissions))))
+                          // set maximum allowed expiryDays to 90
+                          new BitbucketPersonalAccessToken(tokenName, permissions,  ))))
```

Comment:
looks like the parameter is missed here. Will the token be updated periodically e.g. factory / workspace startup?

Reply:
Thanks for the remark, fixed

Reply:
np, but what about the token update, will it be refreshed periodically by design? e.g. what is going to happen after 3 month when the token become invalid

Reply:
We check the token each time when workspace starts or a new one is created. If the token is expired we create a new one.